trafgen.c

```diff
@@ -1056,9 +1058,20 @@ int main(int argc, char **argv)
 			ctx.verbose = true;
 			break;
 		case 'P':
-			cpus_tmp = strtoul(optarg, NULL, 0);
-			if (cpus_tmp > 0 && cpus_tmp < ctx.cpus)
-				ctx.cpus = cpus_tmp;
+			if (slow)
+				break;
+			cpu_n = strtoul(optarg, &ptr, 0);
+			if (ptr && *ptr == '-') {
+				if (cpu_n < 0 || cpu_n >= ctx.cpu_num)
+					break;
+				ctx.cpu_start = cpu_n;
+				ctx.cpu_num -= cpu_n;
+				cpu_n = strtoul(ptr + 1, NULL, 0);
+				if (cpu_n < ctx.cpu_start || cpu_n >= ctx.cpu_num)
+					break;
+				ctx.cpu_num = cpu_n - ctx.cpu_start + 1;
+			} else if (cpu_n > 0 && cpu_n <= ctx.cpu_num)
+				ctx.cpu_num = cpu_n;
```

Comment:
I think something in this logic is not quite right yet. Testing this on a system with 8 CPUs online. For example using `trafgen ... --cpus 2-6` I would expect `cpu_start = 2` and `cpu_num = 5` (CPUs 2, 3, 4, 5, 6), but I get `cpu_num = 6`.

Reply:
Ah, thank you for pointing out. Now it calculates CPUs correctly.